### PR TITLE
Add Better error message for lists in config file

### PR DESCRIPTION
**Pull request by saadmk11**
closes https://github.com/readthedocs/readthedocs.org/issues/4914

### CODE REVIEW

File: readthedocs/config/config.py [hunk 1]
```diff
@@ -110,16 +110,24 @@ class InvalidConfig(ConfigError):
     message_template = 'Invalid "{key}": {error}'
 
     def __init__(self, key, code, error_message, source_file=None):
-        self.key = key
+        self.key = self._get_display_key(key)
```
Comment (stsewd) [inline readthedocs/config/config.py]: The key shouldn't change. The modified key should only be used on the error msg

Comment (saadmk11) [reply readthedocs/config/config.py]: Sure!

File: readthedocs/config/config.py [hunk 2]
```diff
@@ -114,12 +114,20 @@ def __init__(self, key, code, error_message, source_file=None):
...
         self.source_file = source_file
         message = self.message_template.format(
-            key=key,
+            key=self._get_display_key(key),
             code=code,
             error=error_message,
         )
         super().__init__(message, code=code)
 
+    @staticmethod
```
Comment (stsewd) [inline readthedocs/config/config.py]: Any reason to make this a static method? Looks like a normal method works

Comment (saadmk11) [reply readthedocs/config/config.py]: @stsewd if I use `instance` methods then the lint fails with `no-self-use` error. So, I used `staticmethod`. Though we can use `#pylint disable=no-self-use`

Comment (stsewd) [reply readthedocs/config/config.py]: You can use self now `self.key`

Comment (saadmk11) [reply readthedocs/config/config.py]: Done!

**Pull request merged**